src/net/java/sip/communicator/impl/protocol/jabber/OperationSetContactCapabilitiesJabberImpl.java

```diff
@@ -405,45 +444,80 @@ public void userCapsNodeAdded(String user, String node, boolean online)
      * record for a specific user about the caps node the user has.
      *
      * @param user the user (full JID)
+     * @param fullJids a list of all resources of the user (full JIDs)
+     * @param node the entity caps node#ver
+     * @param online indicates if the user is currently online
+     * @see UserCapsNodeListener#userCapsNodeAdded(String, String, boolean)
+     */
+    public void userCapsNodeRemoved(String user, ArrayList<String> fullJids,
+        String node, boolean online)
+    {
+        /*
+         * It doesn't matter to us whether a caps node has been added or removed
+         * for the specified user because we report all changes.
+         */
+        userCapsNodeChanged(user, fullJids, node, online);
+    }
+
+    /**
+     * Notifies this listener that an <tt>EntityCapsManager</tt> has changed a
+     * record for a specific user about the caps node the user has.
+     *
+     * @param user the user (full JID)
+     * @param fullJids a list of all resources of the user (full JIDs)
      * @param node the entity caps node#ver
      * @param online indicates if the given user is online
-     * @see UserCapsNodeListener#userCapsNodeRemoved(String, String, boolean)
      */
-    public void userCapsNodeRemoved(String user, String node, boolean online)
+    public void userCapsNodeChanged(String user, ArrayList<String> fullJids,
+        String node, boolean online)
     {
         OperationSetPresence opsetPresence
-            = parentProvider.getOperationSet(OperationSetPresence.class);
-
-        if (opsetPresence != null)
-        {
-            String jid = StringUtils.parseBareAddress(user);
-            Contact contact = opsetPresence.findContactByID(jid);
-
-            // If the contact isn't null and is online we try to discover the
-            // new set of operation sets and to notify interested parties.
-            // Otherwise we ignore the event.
-            if (contact != null)
+        = parentProvider.getOperationSet(OperationSetPresence.class);
+        if (opsetPresence != null) {
+            if(LibJitsi.getConfigurationService()
+                .getBoolean(
+                    XMPPConfigForm.PROP_XMPP_USE_ALL_RESOURCES_FOR_CAPABILITIES,
+                    XMPPConfigForm.USE_ALL_RESOURCES_FOR_CAPABILITIES_DEFAULT))
```

Comment:
The XMPPConfigForm is in a different bundle and may not be referenced in the protocol-jabber bundle. (It would e.g. break the Videobridge which doesn't have the generalconfig bundle). This probably only works because the constants are expanded at compile time. And because by convention you cannot reference impl-bundles in the UI, I currently only see the option of duplicating those constants in the protocol and the UI bundle.